pkg/proc/gdbserial/gdbserver.go

```diff
@@ -2190,30 +2185,29 @@ func (t *gdbThread) SetPC(pc uint64) error {
 	return t.p.conn.writeRegister(t.strID, reg.regnum, reg.value)
 }
 
-// SetSP will set the value of the SP register to the given value.
-func (t *gdbThread) SetSP(sp uint64) error {
-	_, _ = t.Registers() // Registes must be loaded first
-	t.regs.setSP(sp)
-	if t.p.gcmdok {
-		return t.p.conn.writeRegisters(t.strID, t.regs.buf)
+// ChangeRegs will change the value of a list of registers
+func (t *gdbThread) SetReg(regNum uint64, reg *op.DwarfRegister) error {
+	var regName string
+	switch t.p.bi.Arch.Name {
```

Comment:
Perhaps `Arch` should have a function pointer for translating, like: `Arch.RegisterNumberToName` which behind the scenes will have been set to the correct function from the `regnum` package.

Reply:
LGTM